scripts/simulator.ts

```diff
@@ -0,0 +1,154 @@
+  const shouldRunAll = typeof conf.simulate === "undefined";
+  const shouldRunNone =
+    Array.isArray(conf.simulate) && conf.simulate.length === 0;
+
+  const ensureBotUsers = withErrorReporter(
+    { ...conf.log, critical: true },
+    actualEnsureBotUsers
+  );
+  const userRefs = await ensureBotUsers({ ...options, ...conf.user });
+
+  if (shouldRunNone) {
+    return { userRefs };
+  }
+
+  assert.ok(venue?.id, chalk`${main.name}(): {magenta venue.id} is required`);
+
+  const venueRef = await getVenueRef({ log, venueId: venue.id });
+  const reactionsRef = await getReactionsRef({ venueId: venue.id });
+  const chatsRef = await getChatlinesRef({ venueId: venue.id });
+
+  assert.ok(
+    venueRef,
+    chalk`${main.name}(): venue was not found for {magenta venue.id}: {green ${venue.id}}`
+  );
+
+  const simulatorContext: SimulatorContext = {
+    ...options,
+    userRefs,
+    chatsRef,
+    reactionsRef,
+    venueRef,
+    venueId: venueRef.id,
+  };
+
+  const simulations = [];
+
+  if (shouldRunAll || simulate.includes("chat")) {
+    simulations.push(simChat(simulatorContext));
+  }
+
+  if (shouldRunAll || simulate.includes("experience")) {
+    simulations.push(simExperience(simulatorContext));
+  }
+
+  if (shouldRunAll || simulate.includes("seat")) {
+    simulations.push(simSeat(simulatorContext));
+  }
+
+  await Promise.all(simulations);
+
+  return { chatsRef, reactionsRef, userRefs };
+};
+
+const cleanup: (options: CleanupOptions) => Promise<void> = async ({
+  conf,
+  log,
+  result: { chatsRef, reactionsRef, userRefs },
+  stats,
+}) => {
+  // and now, clean up the mess
+
+  if (conf.user?.cleanup ?? true) {
+    log(chalk`{inverse NOTE} Doing little user cleanup...`);
+    await removeBotUsers({ conf, log, stats, userRefs });
+    log(chalk`{green.inverse DONE} Removed bot users.`);
+  }
+
+  if (reactionsRef && (conf.experience?.cleanup ?? true)) {
+    log(chalk`{inverse NOTE} Doing little reactions cleanup...`);
+    await removeBotReactions({ conf, log, stats, reactionsRef });
+    log(chalk`{green.inverse DONE} Removed bot reactions.`);
+  }
+
+  if (chatsRef && (conf.chat?.cleanup ?? true)) {
+    log(chalk`{inverse NOTE} Doing little reactions cleanup...`);
+    await removeBotChatMessages({ conf, log, stats, chatsRef });
+    log(chalk`{green.inverse DONE} Removed bot reactions.`);
+  }
+
+  const calcPer = (unit: number) => ({
+    writes: unit && stats.writes ? (stats.writes / unit).toFixed(2) : "N/A",
+    relocations:
+      unit && stats.relocations ? (stats.relocations / unit).toFixed(2) : "N/A",
+    reactions:
+      unit && stats.reactions?.created
+        ? (stats.reactions.created / unit).toFixed(2)
+        : "N/A",
+    chatlines:
+      unit && stats.chatlines?.created
+        ? (stats.chatlines.created / unit).toFixed(2)
+        : "N/A",
+  });
+
+  const start = parseISO(stats.time?.start ?? "");
+  const finish = parseISO(stats.time?.finish ?? "");
```

Comment:
NOTE: `parseISO()` from the `date-fns` library converts the time to local time zone (and display the TZ difference) while `Date.prototype.toISOString()` will output it in UTC and attach `Z` at the end.